### DISCUSSION TITLE
Support for bundlers

### DISCUSSION THREAD

**Discussion by azarenkovgd** [kriszyp/lmdb-js#290] [General]
I use rollup to bundle my backend code into one js file, then I deploy it, without any additional node modules. There is a problem with this package - it requires some ".node" dependency (sorry for some wrong terminology, I do now know how to explain it better). Path to this dependency is resolved at a runtime. Seems to be, it happens in the file [https://github.com/kriszyp/lmdb-js/blob/master/native.js](https://github.com/kriszyp/lmdb-js/blob/master/native.js), here:

```ts
import loadNAPI from 'node-gyp-build-optional-packages';

...

let dirName = dirname(fileURLToPath(import.meta.url)).replace(/dist$/, '');
export let nativeAddon = loadNAPI(dirName);
```

Is it possible to allow user to supply path to native addon?

Code could look something like that:

```ts
export let nativeAddon: string;
if (process.env["LMDBJS_NATIVE_ADDON_PATH"]) {
    nativeAddon = require(process.env["LMDBJS_NATIVE_ADDON_PATH"]);
} else {
    let dirName = dirname(fileURLToPath(import.meta.url)).replace(/dist$/, '');
    export let nativeAddon = loadNAPI(dirName);
}
```

I tried studying `node-gyp-build-optional-packages` package, but seems to be, there is no way to solve this problem "official" way. Hope I am wrong.

**Comment by kriszyp** [comment]
I don't know if this issue might have some helpful suggestions:
https://github.com/kriszyp/lmdb-js/issues/278
Also,  there is some logic to look at env variable to find the directory to look for the binary here:
https://github.com/kriszyp/node-gyp-build/blob/master/index.js#L29
Would that help?

**Reply by azarenkovgd** [reply]
Thank you for the response!
It helped
